Doc/library/asyncio-runner.rst

```diff
@@ -75,7 +75,9 @@ Runner context manager
    :ref:`asyncio-debug-mode` settings.
 
    *loop_factory* could be used for overriding the loop creation.
-   :func:`asyncio.new_event_loop` is used if ``None``.
+   It is the responsibility of the *loop_factory* to set the created loop as the
```

Comment:
Who's responsibility is to unset it? Should Runner airways set it to None or should the author of the loop_factory function?

Reply:
The Runner also unsets it when closed/exited. If you feel that this needs to be called out please open a new PR.

Reply:
Runner only unsets it in close if it is the one that sets it.

so the caller of Runner is responsible for unsetting it, and they need to unset it before the loop is closed so the childwatchers that join threads join them before the loop closes?

Reply:
> Runner only unsets it in close if it is the one that sets it.

Which makes sense, right?

> so the caller of Runner is responsible for unsetting it, and they need to unset it before the loop is closed so the childwatchers that join threads join them before the loop closes?

That sounds like a criticism of the code. I don't know enough about child watchers to be able to answer.

Reply:
If you want any further changes, just send a PR.

Reply:
@kumaraditya303 thanks, I will try to do that: I'm actually looking to pretty much write the opposite of what you have here, eg that it's not actually the responsibility for the loop_factory to call `set_event_loop` both because there's no Runner interface to unset it correctly and because asyncio code that doesn't call or rely on `set_event_loop` is easier to get right.

Is that something you'd like to see in a PR?

Reply:
Whatever gives users the best guidance. I searched for code that uses `Runner(loop_factory=...)` and there is one case apart from `test_runners.py`: in `test_asyncio/test_subprocess.py` it us used as `loop_factory=asyncio.new_event_loop` specifically to prevent the watcher to be set (since the test sets its own watcher).

I would emphasize that the default case *does* call `set_event_loop()` and implies attaching the watcher (if there is one!) to the loop; it's fine to suggest that `loop_factory` should be used only if this is undesired.